doc/contributing.rst

```diff
@@ -0,0 +1,90 @@
+Contributing
+============
+
+Before developing features or fixing bugs, please make sure you are have done
+the following:
+
+- Your code is not on the ``main`` branch of your fork.
+- The code has been tested.
+- All commit messages are properly formatted and commits squashed where
+  appropriate.
+- You have included updates to all appropriate documentation.
+
+We use GitHub pull requests to review contributions. Please be kind and patient
+as code reviews can be long and minutious.
+
+
+Development
+-----------
+
+XDG Desktop Portal usually runs as a user session service, initialized on
+demand through D-Bus activation. It usually starts with the session though,
+as many desktop environments try to talk to XDG Desktop Portal on startup.
+XDG Desktop Portal initializes specific backends through D-Bus activation
+as well.
+
+
+Building
```

Comment:
Advising people to install development code into `/usr` isn't exactly a safe thing to do.

Can we come up with a scheme where the developers don't have to modify the host system too much?

There will be cases where you probably want to install to the host `/usr` but in that case at least tell the developer that this is potentially dangerous and maybe even that there are ways to install to the system and then being able to roll back (`ostree admin unlock`, `systemd-sysext`, ...).

Reply:
I merely moved the contents of `CONTRIBUTING.md` into this page. I just moved it here because @hfiguiere suggested in last month's discussion (https://github.com/flatpak/xdg-desktop-portal/discussions/1232#discussioncomment-7708916). I'm not particularly interested in reviewing this page myself right now so patches would be welcomed.